contracts/GNTDeposit.sol

```diff
@@ -126,39 +128,40 @@ contract GNTDeposit is ERC223ReceivingContract {
         balances[msg.sender] = 0;
         locked_until[msg.sender] = 0;
         require(token.transfer(_to, _amount));
-        Withdraw(msg.sender, _to, _amount); // event
+        Withdraw(msg.sender, _to, _amount);
     }
 
     function burn(address _whom, uint256 _burn)
         onlyOracle
         external
-        returns (bool)
     {
-        require(balances[_whom] >= _burn);
-        balances[_whom] -= _burn;
-        if (balances[_whom] == 0) {
-            locked_until[_whom] = 0;
-        }
-        require(token.transfer(0xdeadbeef, _burn));
-        Burn(_whom, _burn); // event
-        return true;
+        _reimburse(_whom, 0xdeadbeef, _burn);
+        Burn(_whom, _burn);
     }
 
-    function reimburse(address _owner, address _receiver, uint256 _reimbursement)
+    function reimburseForSubtask(address _requestor, address _provider, uint256 _amount, string _subtask_id)
         onlyOracle
         external
-        returns (bool)
     {
-        require(balances[_owner] >= _reimbursement);
-        balances[_owner] -= _reimbursement;
-        if (balances[_owner] == 0) {
-            locked_until[_owner] = 0;
-        }
-        require(token.transfer(_receiver, _reimbursement));
-        Reimburse(_owner, _receiver, _reimbursement); // event
-        return true;
+        _reimburse(_requestor, _provider, _amount);
+        ReimburseForSubtask(_requestor, _provider, _amount, _subtask_id);
+    }
+
+    function reimburseForBatchTransfer(address _requestor, address _provider, uint256 _amount, uint256 _closure_time)
```

Comment:
The method name is somehow misleading. I suggest reimburse() or reimburseForNoPayment() or something like this.

Reply:
How about `reimburseForIncorectBatchTransfer`? `reimburse` alone is not descriptive enough compared to other `reimburse` methods and `reimburseForNoPayment` may not be precise in case of partial payments.

Reply:
I'll go with `NoPayment` after all.